File: Add Code Here/ReverseLinkedList.java
import java.util.*;

public class ReverseLinkedList {
    public static Lcheck head = null;
    public static class Lcheck {
        int data;
        Lcheck next;
    }
    public static void get(int value) {
        Lcheck temp = new Lcheck();
        temp.data = value;
        temp.next = null;
        if (head == null) {
            head = temp;
        }
        else {
            Lcheck last_check = head;
            while (last_check.next != null) {
                last_check = last_check.next;
            }
            last_check.next = temp;
        }
    }
    public static void print(Lcheck temp) {
        while (temp != null) {
            System.out.print(temp.data + " ");
            temp = temp.next;
        }
    }

    public static void reverseLinkedlist() {

        Stack<Lcheck> t = new Stack<>();
        Lcheck temp = head;
        while (temp.next != null) {
            t.add(temp);
            temp = temp.next;
        }
        head = temp;
        while (!t.isEmpty()) {
            temp.next = t.peek();
            t.pop();
            temp = temp.next;
        }
        temp.next = null;
    }
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);

        System.out.println("Enter 5 elements  for Reverse linked list");
        for (int i = 0; i < 5; i++) {
            int e = in.nextInt();
            get(e);
        }
        System.out.print("linked list Before Reverse\n");
        print(head);
        reverseLinkedlist();
        System.out.print("\nlinked list After Reverse\n");
        print(head);
    }
}

File: Add Code Here/C++/reverse_linked_list.cpp
#include <iostream>
#include <vector>

using namespace std;

class ListNode{
    public:
    int data;
    ListNode* next;

    ListNode(int val){
        data = val;
        next = NULL;
    }
};

ListNode* reverse(ListNode* head){
    ListNode* prev = NULL;
    ListNode* curr = head;
    ListNode* next = NULL;
    while(curr != NULL){
        next = curr->next;
        curr->next = prev;

        prev = curr;
        curr = next;
    }
    return prev;
}

int main(){
    ListNode* head = new ListNode(1);
    head->next = new ListNode(2);
    head->next->next = new ListNode(3);

    ListNode* newHead = reverse(head);
    while(newHead != NULL){
        cout<<newHead->data<<" ";
        newHead = newHead->next;
    }
    cout<<endl;
    return 0;
}